src/components/ImportOnyxState/index.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import Onyx from 'react-native-onyx';
+import type {FileObject} from '@components/AttachmentModal';
+import AttachmentPicker from '@components/AttachmentPicker';
+import * as Expensicons from '@components/Icon/Expensicons';
+import MenuItem from '@components/MenuItem';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {KEYS_TO_PRESERVE, setIsUsingImportedState} from '@libs/actions/App';
+import {setShouldForceOffline} from '@libs/actions/Network';
+import Navigation from '@libs/Navigation/Navigation';
+import type {OnyxValues} from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type ImportOnyxStateProps from './types';
+import {cleanAndTransformState} from './utils';
+
+export default function ImportOnyxState({setIsLoading, isLoading}: ImportOnyxStateProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+
+    const handleFileRead = (file: FileObject) => {
+        if (!file.uri) {
+            return;
+        }
+
+        setIsLoading(true);
+        const blob = new Blob([file as BlobPart]);
+        const response = new Response(blob);
+
+        response.text().then((text) => {
+            const fileContent = text;
+            const transformedState = cleanAndTransformState<OnyxValues>(fileContent);
+            setShouldForceOffline(true);
+            Onyx.clear(KEYS_TO_PRESERVE).then(() => {
+                Onyx.multiSet(transformedState)
+                    .then(() => {
+                        setIsUsingImportedState(true);
+                        Navigation.navigate(ROUTES.HOME);
+                    })
+                    .finally(() => {
+                        setIsLoading(false);
+                    });
+            });
+        });
```

Comment:
Let's add a catch block to handle exceptional (same to native file)

Reply:
In case users upload wrong files, we can display a modal to announce that the upload failed:

Similar to : https://github.com/Expensify/App/blob/bc106fab216fbf60cd55004a94223cacf5afa8d9/src/pages/workspace/categories/WorkspaceCategoriesPage.tsx#L432

Reply:
that's a good idea, I have added the modal 👍